### PR TITLE
[SYCL][Driver] Link Asan device lib when enabling Asan via Xarch_device

### PR DESCRIPTION
**Pull request by jinge90**
User can enable Asan for SYCL device code via -Xarch_device, we need to link Asan device library with user's device image in such case.

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @mdtoguchi 
Could you help review this patch?
Thanks very much.

File: clang/test/Driver/sycl-device-lib.cpp [hunk 1]
```diff
@@ -220,6 +220,10 @@
 // RUN:   | FileCheck %s -check-prefix=SYCL_DEVICE_LIB_SANITIZER
 // RUN: %clangxx -fsycl %s --sysroot=%S/Inputs/SYCL -Xsycl-target-frontend=spir64 -fsanitize=address -### 2>&1 \
 // RUN:   | FileCheck %s -check-prefix=SYCL_DEVICE_LIB_SANITIZER
+// RUN: %clangxx -fsycl %s --sysroot=%S/Inputs/SYCL -Xarch_device -fsanitize=address -### 2>&1 \
+// RUN:   | FileCheck %s -check-prefix=SYCL_DEVICE_LIB_SANITIZER
+// RUN: %clangxx -fsycl %s --sysroot=%S/Inputs/SYCL -Xarch_device "-fsanitize=address -DUSE_SYCL_DEVICE_ASAN" -### 2>&1 \
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-device-lib.cpp:225]: Should also check the device `-cc1` line for the macro to make sure it is passed as expected.

Comment (jinge90) [reply clang/test/Driver/sycl-device-lib.cpp:225]: Hi, @mdtoguchi 
Done.
Thanks very much.

**Review by mdtoguchi — APPROVED** [review]
LGTM - Thanks!

**Comment by jinge90** [comment]
Hi, @intel/llvm-gatekeepers 
Could you help merge this patch?
Thanks very much.

**Pull request merged**